### DISCUSSION TITLE
Sound Quality

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/grandorgue#691] [General]
When developing software the likes of those that are used for playing virtual instruments, realism is what one wants to achieve. 

The latest has been the developments in HW VI in which, according to what I briefly read, was the improvements made to how audio is processed and outputted. I am not so technical and I don't understand the meaning or the significance of the lingo. Essentially is seems to included the possibility of increasing the capability of the software to output sound at a higher definition 96kHz sample-rate (see https://forum.hauptwerk.com/viewtopic.php?f=1&t=19209). Apparently, this is associated with lower polyphony needed and therefore reducing the burden on memory resources of the computer hardware being used. There is another feature that has been introduced: "real-time audio pitch shifting".

The latter HW change possibly is the same or similar to that effected in jOrgan by Paul Stratman a long time ago. He called it "flexible (variable) wind". This added to the realism experience. However, this feature is _not_ inherent to (coded into) the software but more to specific sample-sets where it is embedded in the respective odf, a feature that can be turned on or off from the virtual console. I imagine that embedding the feature into the actual software widens its usefulness allowing all sample-sets to avail of it not just those that have it coded within their odf. This is where I think the difference lies between jO and HW VI in reference to this feature.

In GrandOrgue, we have often associated sound improvements with the use of so-called back-ends (developed by Graham Goode) &/or the use of ASIO as the audio engine. ASIO could never be incorporated into GO because of legal restrictions apparently. So much so, ASIO is no longer distributed with GO much to the consternation of many.

I think it may be worth exploring the innovations introduced in jO and HW and incorporate them into GO.

I hope that what I wrote makes sense and considered a worthy contribution in the further development of this marvelous software we are using. GO continues to be my preferred choice and I recommend it to anyone without the least hesitation!

Mark

**Comment by Bumblebee001** [comment]
Reading more about HW's recent audio improvements, it appears that of the two features, real-time pitch shifting is of more significance than selecting 96kHz sample-rate.

**Comment by oleg68** [comment]
Thanks, @Bumblebee001 for your thouths.

Could you explain what are you talking about when you mention 96kHz sample-rate? GrandOrgue already supports it. Or you are talking about 192?

**Comment by oleg68** [comment]
> In GrandOrgue, we have often associated sound improvements with the use of so-called back-ends (developed by Graham Goode) 

JOrgan does not produce sound itself and use backends for it. But GrandOrgue has its own sound engine and does not use any backends.

Or you are talking about sound ports (PortAudio, RtAudio with several APIs, Jack)? But they should not take audio quality and only influance on the latency, stability and CPU load.

> &/or the use of ASIO as the audio engine.
ASIO also should not touch the sound quality. If you mean that it eliminates resampling, then there are oters windows sound APIs that also  output without resampling.

From my point of view ASIO influences only the latency.

>  ASIO could never be incorporated into GO because of legal restrictions apparently. So much so, ASIO is no longer distributed with GO much to the consternation of many.

I'm not a lawer, but ASIO SDK may be downloaded from the Steinberg site, and the license agreement does not prevent downloading it and compilation of third party code against it. So my github windows releases come with ASIO support enabled. I would disable it if some lawer told me that it is illegal.

**Reply by Bumblebee001** [reply]
> JOrgan does not produce sound itself

You are right. It goes to show how limited the accuracy of my knowledge is. I know that jO provide a "Dsound" option besides "PortAudio" when installed.

> ASIO

I stated what I stated from what I read / understood in past forum replies applicable at least until that time. Maybe there were restrictions but since removed. I am not a lawyer either!

Regards

**Comment by Bumblebee001** [comment]
Apparently, if one were to be able to select audio output at that frequency i.e. 96 kHz, (and I suppose it also depends on the soundcard/s capability), irrespective of what frequency the actual samples are recorded at or finally processed at, the quality is supposed to improve, though this comes at a cost of processing power by computer. Any higher rate than that will not be detected by our ears so anything above 96 kHz is useless and will only consume more computing power needlessly. Previous editions of HW, from what I understand, selection of such a setting was not possible; with HW VI it is possible and people have reported a marked improvement on this basis alone.

I record my samples at 96kHz but down size to 48 kHz during processing. 48kHz is DVD quality; 44.1 kHz is general CD quality audio. (please correct me if I am wrong).

If GO already provides this general setting for Audio output, then I guess it has already been taken care of. What is more relevant and useful would be the other feature, also made available in HW VI, that of "real-time audio pitch shifting" which has been available for jOrgan until HW VI came along (assuming that this is the same as "flexible wind").

Mark

**Comment by Bumblebee001** [comment]
I must apologize for writing certain things without reviewing the software leading to inaccuracies. I wrote off-the-cuff from what I could recall. I admit that I have taken a rather longish break from using the software and playing VPOs. I do try to keep abreast of developments by quickly reading through some posts largely coming through as emails. At the moment I am much more deeply committed in my professional duties in local health care (pandemic and all) whilst inching closer to my retirement due in just over a year from now. A number of personal issues absorbed a significant amount of my mental energy. 

The bottom line: I was very pleased to learn of the initiative being undertaken by Oleg and wanted to help by drawing attention to areas that might require some attention. I have wrongfully attributed G Goode's back-ends to GO when it should have been jO. Sorry Graham, and all. Also I confirm that GO does have the 96 KHz output option (there are some things I never bothered to touch or explore). I think the rest of my comments are more or less correct and valid.

Best regards to all.

**Comment by oleg68** [comment]
I up this topic

**Comment by rousseldenis** [comment]
My two cents on ASIO, I've read on Steinberg site that the librairies can be used in third-party software as soon people sign agreement.

I think this can be achieved doing this in the name of GrandOrgue (maybe with an explicit agreement of Steinberg as GrandOrgue has no legal statements - but maybe can in a near future, that's another debate).

https://www.steinberg.net/fr/developers/
